packages/grafana-ui/src/components/Button/Button.tsx

```diff
@@ -9,11 +9,14 @@ import { getFocusStyles, getMouseFocusStyles } from '../../themes/mixins';
 import { Icon } from '../Icon/Icon';
 
 export type ButtonVariant = 'primary' | 'secondary' | 'destructive' | 'link';
-export const allButtonVariants: ButtonVariant[] = ['primary', 'secondary', 'destructive', 'link'];
+export const allButtonVariants: ButtonVariant[] = ['primary', 'secondary', 'destructive'];
+export type ButtonStyle = 'solid' | 'outline' | 'text';
+export const allButtonStyles: ButtonStyle[] = ['solid', 'outline', 'text'];
 
 type CommonProps = {
   size?: ComponentSize;
   variant?: ButtonVariant;
+  buttonStyle?: ButtonStyle;
```

Comment:
- `style`
- `as`
- `kind`
- `mode`

Somewhat `style` feels best too me

Reply:
I would probably go for `kind`. I would assume style means `style` like an inline css that will be passed to `<button/>` while `mode` to me evokes change in behavior rather than visual changes. Not really strong feelings though as looking at the types probably makes it quick to find out.